libbeat/api/routes.go

```diff
@@ -73,3 +79,8 @@ func prettyPrint(w http.ResponseWriter, data common.MapStr, u *url.URL) {
 		fmt.Fprintf(w, data.String())
 	}
 }
+
+// AddHandlerFunc provides interface to add customized handlerFunc
+func AddHandlerFunc(api string, h handlerFunc) {
+	handlerFuncMap[api] = h
```

Comment:
We should error/panic if the api is already in use. A new handlerFunc should not be allowed to replace an older handlerFunc.

Reply:
This is now considered in the new implementation.

Reply:
reverted to previous implementation and added check in this function..